### DISCUSSION TITLE
Is there any component that displays based on byte stream? Or implement ideas?

### DISCUSSION THREAD

**Discussion by peakedshout** [fyne-io/fyne#4617] [Q&A]
Currently, I need to display log information on the page. But my log information is a byte stream. 
I need to close the byte stream when the user leaves the page, such as the AppTabs component, and the OnUnselected callback will not be triggered when the user leaves the page.

**Comment by andydotxyz** [comment]
> and the OnUnselected callback will not be triggered when the user leaves the page.

The `OnUnselected` is called when the user changes tab away from the current tab.
I am unsure if this is what you mean - we don't really work with "pages" in a GUI toolkit.

**Reply by peakedshout** [reply]
This means that it will only be triggered if the user selects another tab in this component, and if the user skips the tab directly through the other component, it won't be triggered, right?

**Reply by andydotxyz** [reply]
I don't understand. If the user "skips the tab" then it won't have been shown so the stream won't be open anyway?

**Reply by peakedshout** [reply]
No, it's not how to open it, it's a problem that no one closes it when the stream is turned on.

**Reply by andydotxyz** [reply]
Yes, but if the tab /has/ been opened then it will use the stream to show data - then the user cannot go to another tab and skip the one that is open can they?

**Reply by peakedshout** [reply]
Similar. I'm envisioning that the stream is closed as long as the user isn't on this tab. I don't think the experience would be great if the user was forced to have to end the stream on this tab.
Users can jump to all available components.

**Reply by andydotxyz** [reply]
Yes, and that is what `OnUnselected` is for - you will be told when the user has left that tab.

**Reply by peakedshout** [reply]
However, if the user navigates away from the tab page by clicking on another tab (e.g., by clicking on another tab at a higher level), the callback is not triggered.

**Reply by andydotxyz** [reply]
Yes, If your application wants to handle navigation at multiple levels you will need to put in some more code.
